src/Notifier.js

```diff
@@ -133,29 +153,35 @@ var Notifier = {
         }
 
         if(enable) {
-            if (!this.havePermission()) {
-                global.Notification.requestPermission(function() {
-                    if (callback) {
-                        callback();
+            // Case when we do not have the permission as 'granted'
+            if (this.isPermissionDefault()) {
+                // Attempt to get permission from user
+                var _this = this;
+                global.Notification.requestPermission().then(function(result) {
+                    if (result === 'denied') {
                         dis.dispatch({
                             action: "notifier_enabled",
-                            value: true
+                            value: false
                         });
+                        _this.setToolbarHidden(true);
```

Comment:
I also don't think it's correct to hide the toolbar when the user clicks 'block' on the dialog.